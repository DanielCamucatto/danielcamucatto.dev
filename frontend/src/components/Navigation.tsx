
import { useLanguage } from '../hooks/useLanguage';
import { getTranslations } from '../utils/translations';
import { useSmoothScroll } from '../hooks/useSmoothScroll';
import { useActiveSection } from '../hooks/useActiveSection';

interface NavigationTranslations {
  about: string;
  experience: string;
  projects: string;
  education: string;
}

export default function Navigation() {
  const { language } = useLanguage();
  const translations = getTranslations(language).navigation as NavigationTranslations;
  const { handleNavClick } = useSmoothScroll();
  const activeSection = useActiveSection();

  return (
    <nav className="nav hidden lg:block" aria-label="In-page jump links">
      <ul className="mt-8 w-max">
        <li>
          <a 
            className={`group flex items-center py-3 ${activeSection === 'about' ? 'active' : ''}`}
            href="#about"
            onClick={(e) => handleNavClick(e, 'about')}
          >
            <span className={`nav-indicator mr-4 h-px transition-all motion-reduce:transition-none ${
              activeSection === 'about' 
                ? 'w-16 bg-slate-800 dark:bg-slate-200' 
                : 'w-8 bg-slate-400 dark:bg-slate-600 group-hover:w-16 group-hover:bg-slate-800 dark:group-hover:bg-slate-200 group-focus-visible:w-16 group-focus-visible:bg-slate-800 dark:group-focus-visible:bg-slate-200'
            }`}></span>
            <span className={`nav-text text-xs font-bold uppercase tracking-widest transition-colors ${
              activeSection === 'about'
                ? 'text-slate-900 dark:text-slate-200'
                : 'text-slate-600 dark:text-slate-500 group-hover:text-slate-900 dark:group-hover:text-slate-200 group-focus-visible:text-slate-900 dark:group-focus-visible:text-slate-200'
            }`}>
              {translations.about}
            </span>
          </a>
        </li>
        <li>
          <a 
            className={`group flex items-center py-3 ${activeSection === 'experience' ? 'active' : ''}`}
            href="#experience"
            onClick={(e) => handleNavClick(e, 'experience')}
          >
            <span className={`nav-indicator mr-4 h-px transition-all motion-reduce:transition-none ${
              activeSection === 'experience' 
                ? 'w-16 bg-slate-800 dark:bg-slate-200' 
                : 'w-8 bg-slate-400 dark:bg-slate-600 group-hover:w-16 group-hover:bg-slate-800 dark:group-hover:bg-slate-200 group-focus-visible:w-16 group-focus-visible:bg-slate-800 dark:group-focus-visible:bg-slate-200'
            }`}></span>
            <span className={`nav-text text-xs font-bold uppercase tracking-widest transition-colors ${
              activeSection === 'experience'
                ? 'text-slate-900 dark:text-slate-200'
                : 'text-slate-600 dark:text-slate-500 group-hover:text-slate-900 dark:group-hover:text-slate-200 group-focus-visible:text-slate-900 dark:group-focus-visible:text-slate-200'
            }`}>
              {translations.experience}
            </span>
          </a>
        </li>
        <li>
          <a 
            className={`group flex items-center py-3 ${activeSection === 'projects' ? 'active' : ''}`}
            href="#projects"
            onClick={(e) => handleNavClick(e, 'projects')}
          >
            <span className={`nav-indicator mr-4 h-px transition-all motion-reduce:transition-none ${
              activeSection === 'projects' 
                ? 'w-16 bg-slate-800 dark:bg-slate-200' 
                : 'w-8 bg-slate-400 dark:bg-slate-600 group-hover:w-16 group-hover:bg-slate-800 dark:group-hover:bg-slate-200 group-focus-visible:w-16 group-focus-visible:bg-slate-800 dark:group-focus-visible:bg-slate-200'
            }`}></span>
            <span className={`nav-text text-xs font-bold uppercase tracking-widest transition-colors ${
              activeSection === 'projects'
                ? 'text-slate-900 dark:text-slate-200'
                : 'text-slate-600 dark:text-slate-500 group-hover:text-slate-900 dark:group-hover:text-slate-200 group-focus-visible:text-slate-900 dark:group-focus-visible:text-slate-200'
            }`}>
              {translations.projects}
            </span>
          </a>
        </li>
        <li>
          <a 
            className={`group flex items-center py-3 ${activeSection === 'education' ? 'active' : ''}`}
            href="#education"
            onClick={(e) => handleNavClick(e, 'education')}
          >
            <span className={`nav-indicator mr-4 h-px transition-all motion-reduce:transition-none ${
              activeSection === 'education' 
                ? 'w-16 bg-slate-800 dark:bg-slate-200' 
                : 'w-8 bg-slate-400 dark:bg-slate-600 group-hover:w-16 group-hover:bg-slate-800 dark:group-hover:bg-slate-200 group-focus-visible:w-16 group-focus-visible:bg-slate-800 dark:group-focus-visible:bg-slate-200'
            }`}></span>
            <span className={`nav-text text-xs font-bold uppercase tracking-widest transition-colors ${
              activeSection === 'education'
                ? 'text-slate-900 dark:text-slate-200'
                : 'text-slate-600 dark:text-slate-500 group-hover:text-slate-900 dark:group-hover:text-slate-200 group-focus-visible:text-slate-900 dark:group-focus-visible:text-slate-200'
            }`}>
              {translations.education}
            </span>
          </a>
        </li>
      </ul>
    </nav>
  );
}
