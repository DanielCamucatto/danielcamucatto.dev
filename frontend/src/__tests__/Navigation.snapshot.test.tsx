import { render, screen } from '@testing-library/react';
import Navigation from '../components/Navigation';

describe('Navegação', () => {
  it('renderiza todos os links de navegação', () => {
    render(<Navigation />);
    expect(screen.getByRole('link', { name: /sobre/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /experi/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /projeto/i })).toBeInTheDocument();
  });

  it('links de navegação apontam para as seções corretas', () => {
    render(<Navigation />);
    expect(screen.getByRole('link', { name: /sobre/i })).toHaveAttribute('href', '#about');
    expect(screen.getByRole('link', { name: /experi/i })).toHaveAttribute('href', '#experience');
    expect(screen.getByRole('link', { name: /projeto/i })).toHaveAttribute('href', '#projects');
  });

  it('snapshot da navegação', () => {
    const { asFragment } = render(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });
});
