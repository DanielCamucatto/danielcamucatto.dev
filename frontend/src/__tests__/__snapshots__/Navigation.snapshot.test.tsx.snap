// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Navigation snapshot snapshot da navegação 1`] = `
<DocumentFragment>
  <nav
    aria-label="In-page jump links"
    class="nav hidden lg:block"
  >
    <ul
      class="mt-8 w-max"
    >
      <li>
        <a
          class="group flex items-center py-3 active"
          href="#about"
        >
          <span
            class="nav-indicator mr-4 h-px transition-all motion-reduce:transition-none w-16 bg-slate-800 dark:bg-slate-200"
          />
          <span
            class="nav-text text-xs font-bold uppercase tracking-widest transition-colors text-slate-900 dark:text-slate-200"
          >
            Sobre
          </span>
        </a>
      </li>
      <li>
        <a
          class="group flex items-center py-3 "
          href="#experience"
        >
          <span
            class="nav-indicator mr-4 h-px transition-all motion-reduce:transition-none w-8 bg-slate-400 dark:bg-slate-600 group-hover:w-16 group-hover:bg-slate-800 dark:group-hover:bg-slate-200 group-focus-visible:w-16 group-focus-visible:bg-slate-800 dark:group-focus-visible:bg-slate-200"
          />
          <span
            class="nav-text text-xs font-bold uppercase tracking-widest transition-colors text-slate-600 dark:text-slate-500 group-hover:text-slate-900 dark:group-hover:text-slate-200 group-focus-visible:text-slate-900 dark:group-focus-visible:text-slate-200"
          >
            Experiência
          </span>
        </a>
      </li>
      <li>
        <a
          class="group flex items-center py-3 "
          href="#projects"
        >
          <span
            class="nav-indicator mr-4 h-px transition-all motion-reduce:transition-none w-8 bg-slate-400 dark:bg-slate-600 group-hover:w-16 group-hover:bg-slate-800 dark:group-hover:bg-slate-200 group-focus-visible:w-16 group-focus-visible:bg-slate-800 dark:group-focus-visible:bg-slate-200"
          />
          <span
            class="nav-text text-xs font-bold uppercase tracking-widest transition-colors text-slate-600 dark:text-slate-500 group-hover:text-slate-900 dark:group-hover:text-slate-200 group-focus-visible:text-slate-900 dark:group-focus-visible:text-slate-200"
          >
            Projetos
          </span>
        </a>
      </li>
      <li>
        <a
          class="group flex items-center py-3 "
          href="#education"
        >
          <span
            class="nav-indicator mr-4 h-px transition-all motion-reduce:transition-none w-8 bg-slate-400 dark:bg-slate-600 group-hover:w-16 group-hover:bg-slate-800 dark:group-hover:bg-slate-200 group-focus-visible:w-16 group-focus-visible:bg-slate-800 dark:group-focus-visible:bg-slate-200"
          />
          <span
            class="nav-text text-xs font-bold uppercase tracking-widest transition-colors text-slate-600 dark:text-slate-500 group-hover:text-slate-900 dark:group-hover:text-slate-200 group-focus-visible:text-slate-900 dark:group-focus-visible:text-slate-200"
          >
            Formação
          </span>
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
